lib/ansible/modules/cloud/amazon/GUIDELINES.md

```diff
@@ -146,10 +147,10 @@ exception handling like in boto.  Instead, an `AuthFailure` exception will be th
 
 ```python
 region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
-if region:
+try:
     connection = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_params)
```

Comment:
sorry this might be a silly question, but why can't we catch the `NoRegionError` in the `boto3_conn` function and fail the task? Wouldn't that be easier as this check won't need to happen in individual modules?

Reply:
It's not a silly question at all

One answer is that route53, cloudfront, iam etc. don't need a region.

But that's not to say we couldn't pass a `check_region` parameter to `boto3_conn` and handle the problem centrally.